tests/common/utilities.py

```diff
@@ -1112,18 +1112,31 @@ def capture_and_check_packet_on_dut(
         duthost.file(path=pcap_save_path, state="absent")
 
 
-def duthost_ssh(duthost, sonic_username, sonic_passwords, sonic_ip):
-    for password in sonic_passwords:
+def paramiko_ssh(username, passwords, ipaddr):
+    if type(passwords) == str:
         try:
             ssh = paramiko.SSHClient()
             ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-            ssh.connect(sonic_ip, username=sonic_username, password=password,
+            ssh.connect(ipaddr, username=username, password=passwords,
                         allow_agent=False, look_for_keys=False, timeout=10)
             return ssh
-        except paramiko.AuthenticationException:
-            continue
         except Exception as e:
-            logging.info("Cannot access DUT {} via ssh, error: {}".format(duthost.hostname, e))
             raise e
-    logging.info("Cannot access DUT {} via ssh, error: Password incorrect".format(duthost.hostname))
-    raise paramiko.AuthenticationException
+
+    elif type(passwords) == list:
+        for password in passwords:
+            try:
+                ssh = paramiko.SSHClient()
+                ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+                ssh.connect(ipaddr, username=username, password=password,
+                            allow_agent=False, look_for_keys=False, timeout=10)
```

Comment:
There is code repetition here. It would be easier to workaround it with code like below:
```python
if isinstance(passwords, str):
    candidate_passwords = [passwords]
elif isinstance(passwords, list):
    candidate_passwords = passwords
else:
    raise Exception("The passwords argument must be either a string or a list of string.") 

# Then loop through the candidate_passwords to try connection.

```

Reply:
Good idea!